src/buffer_server.cc

```diff
@@ -31,6 +33,15 @@ Server::Server(const ServerOptions& options)
       kLookupTransformServiceName, &Server::handleLookupTransform, this));
   service_servers_.push_back(private_node_handle.advertiseService(
       kCanTransformServiceName, &Server::handleCanTransform, this));
+  if (options_.add_legacy_server) {
+    optional_legacy_server_ = std::make_unique<tf2_ros::BufferServer>(
+        std::cref(tf_buffer_),
+        options_.legacy_server_namespace.empty()
```

Comment:
Would it make sense to add this logic to a function and also log the name of the legacy server in server_main line 78?

Reply:
You're right, it doesn't make sense to decide this here. I moved it to the command line handling in the main function, see 6f3562b3c87443c135c6074e0fbee966ad870140
I don't think a dedicated function is needed for this.